addons/web/static/src/js/fields/basic_fields.js

```diff
@@ -2599,6 +2599,15 @@ var StateSelectionWidget = AbstractField.extend({
             .parent().attr('title', currentState.state_name)
             .attr('aria-label', this.string + ": " + currentState.state_name);
 
+        // Add label with status icon
+        if (this.nodeOptions && this.nodeOptions.show_label) {
+            const $label = $('<span>', {
+                text: currentState.state_name,
+                class: 'align-middle',
+            });
+            this.$el.append($label);
+        }
+
```

Comment:
Hello @mba-odoo 🙂 
As per the JS team requirement, we have to create a JS test in order to ensure that this gets not broken in the future. We will also need to update the documentation. I'll ping them in the PR.

Reply:
@IT-Ideas , Done the changes. Thanks